lsp-clients.el

```diff
@@ -828,5 +830,120 @@ responsiveness at the cost of possible stability issues."
+
+(defun parse-rf-language-server-library-dirs (dirs)
+  (mapcar
+   (lambda (x)
+     (concat
+      (expand-file-name
+       lsp-rf-language-server-dir)
+      x
+      ".json"))
+   dirs))
+
+(defun parse-rf-language-server-globs-to-regex (vector)
+  "Converts vector with globs to regex"
+    (concat "\\("(mapconcat 'eshell-glob-regexp vector "\\|") "\\)"))
+
+(defun parse-rf-language-server-include-path-regex (vector)
+  "Creates regexp to select files from workspace directory"
+  (let ((globs (if (eq vector [])
+                        ["*.robot" "*.resource"]
+                      vector)))
+    (parse-rf-language-server-globs-to-regex globs)))
+
+(defun parse-rf-language-server-exclude-paths (seq)
+  "Creates regexp to select files from workspace directory"
+  (if (eq lsp-rf-language-server-exclude-paths [])
+      seq
+  (cl-delete-if (lambda (x) (string-match-p
+                             (parse-rf-language-server-globs-to-regex
+                              lsp-rf-language-server-exclude-paths)
+                             x))
+                seq)))
+
+(lsp-register-custom-settings
+ '(
+   ("rfLanguageServer.trace.server" lsp-rf-language-server-trace-server)
+   ("rfLanguageServer.logLevel" lsp-rf-language-server-log-level)
+   ("rfLanguageServer.libraries" lsp-rf-language-server-libraries)
+   ("rfLanguageServer.pythonKeywords" lsp-rf-language-server-python-keywords t)
+   ("rfLanguageServer.excludePaths" lsp-rf-language-server-exclude-paths)
+   ("rfLanguageServer.includePaths" lsp-rf-language-server-include-paths)
+   ))
+
+(lsp-register-client
+ (make-lsp-client :new-connection (lsp-stdio-connection
+                                   `("node" ,(expand-file-name
+                                              lsp-rf-language-server-main-file)))
+                  :major-modes '(robot-mode)
+                  :server-id 'rf-intellisense
+                  ;; :library-folders-fn (lambda (_workspace)
+                  ;;                        lsp-rf-language-server-libraries)
+                  :library-folders-fn (lambda (_workspace)
+                                         (parse-rf-language-server-library-dirs
+                                         lsp-rf-language-server-libraries))
+                  :initialized-fn (lambda (workspace)
+                                    (with-lsp-workspace workspace
+                                      (lsp--set-configuration
+                                       (lsp-configuration-section "rfLanguageServer"))
+                                      (lsp-request "buildFromFiles"
+                                                   (list :files
+                                                         (vconcat
+                                                          (parse-rf-language-server-exclude-paths
+                                                           (directory-files-recursively
+                                                            (lsp-workspace-root)
```

Comment:
pass `workspace`

Reply:
What do you mean? You mean pass that to the `lsp--workspace-root`? (Probably not because that doesn't work)

Reply:
should be `(lsp--workspace-root workspace)`

Reply:
Okay, replaced that line. However, I tested this before from a buffer that should be "inside" the workspace but then workspace was void. Now I see the workspace is passed by the initialized-fn "keyword".